ios/RNTrackPlayer/RNTrackPlayer.swift

```diff
@@ -130,11 +133,72 @@ public class RNTrackPlayer: RCTEventEmitter, AudioPlayerDelegate {
     
     @objc(setupPlayer:resolver:rejecter:)
     public func setupPlayer(config: [String: Any], resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
-        do {
-            try AVAudioSession.sharedInstance().setCategory(.playback, mode: .default)
-            try AVAudioSession.sharedInstance().setActive(true)
-        } catch {
-            reject("setup_audio_session_failed", "Failed to setup audio session", error)
+        
+        let sessionCategory: String = config["iosCategory"] as? String ?? "playback"
```

Comment:
another comment: this has all gotten quite big. Let's take the approach we've taken for  `Capabilities` & `PitchAlgorithms` on these.

You can see the examples here: https://github.com/react-native-kit/react-native-track-player/tree/dev/ios/RNTrackPlayer/Models